### PR TITLE
minor improvements

### PR DESCRIPTION
**Pull request by antgonza**
There are 2 main minor improvements:
- description should be required; it has always been required but I guess we missed before because we were adding it via the QIIME 1 mapping file, which we are not creating anymore
- simplifying code and adding some support for job arrays in qiita-recover-jobs

### CODE REVIEW

File: scripts/qiita-recover-jobs [hunk 1]
```diff
@@ -41,27 +42,38 @@ def _submit_jobs(jids_to_recover, recover_type):
...
 
 def _retrieve_queue_jobs():
-    qiita_jobs = [line.split()[0]
-                  for line in check_output("qstat").decode('ascii').split("\n")
-                  # just retriving 'qiita' and ignoring [] (ipython workers)
-                  if 'qiita' in line and '[]' not in line and
-                  # and private jobs
-                  'private' not in line and
-                  'STDIN' not in line]
+    lines = check_output(["qstat", "-f"]).decode('ascii').split("\n")
```
Comment (ElDeveloper) [inline scripts/qiita-recover-jobs:45]: This is minor but it looks like the code in this function could be simplified if the jobs output was parsed into a table with a column for job owner, job name, etc.

File: scripts/qiita-recover-jobs [hunk 2]
```diff
@@ -41,27 +42,38 @@ def _submit_jobs(jids_to_recover, recover_type):
...
     return set(qiita_jids)
 
 
+def _count_jobs_in_arrays():
+    # a direct way to calculate how many jobs are in the queue is to count
+    # how many samples are in the preparation information file of the input
+    # artifact
+    jobs = [ProcessingJob(jid) for jid in _retrieve_queue_jobs()
+            if ProcessingJob(jid).command.name in ARRAY_COMMANDS]
+    return sum([len(job.input_artifacts[0].prep_templates[0]) for job in jobs])
```
Comment (ElDeveloper) [inline scripts/qiita-recover-jobs:74]: I thought jobs were processed in batches rather than 1 sample = 1 job. Is that not the case?

**Review by ElDeveloper — APPROVED** [review]
Looks good overall, just two questions.

File: scripts/qiita-recover-jobs (hunk 1 above)
Comment (antgonza) [reply scripts/qiita-recover-jobs:45]: That's correct and thought about it ... my only concern (and reason I didn't implement like that) is that we will be creating a full parser for the output, which IMOO will add complexity. Note that this is the second iteration of this block and works fine as it is; however, we _might_ need to implement the table suggestion in a following version; just not sure right now.

File: scripts/qiita-recover-jobs (hunk 2 above)
Comment (antgonza) [reply scripts/qiita-recover-jobs:74]: Not for job arrays, that's the "feature" it submits one job (array) with multiple internal jobs. The issue and why this function is useful, is that with our current scheduler those internal jobs are actually seen as individual jobs and if we have more than ~4K jobs in the queue the scheduler will stop reading any (new) jobs below that number. Fun times!

**Review by antgonza — COMMENTED** [review]
@ElDeveloper, thank you for the insightful comments.

**Comment by ElDeveloper** [comment]
Thanks!

**Pull request merged**